src/decode/lzma.rs

```diff
@@ -97,10 +98,12 @@ impl LZMAParams {
     }
 }
 
-pub struct DecoderState<LZB>
+pub struct DecoderState<W, LZB>
 where
-    LZB: lzbuffer::LZBuffer,
+    W: io::Write,
+    LZB: lzbuffer::LZBuffer<W>,
 {
+    _phantom: PhantomData<W>,
```

Comment:
I'm not sure I understand why this `PhantomData` is needed. Could you briefly explain?

Reply:
It's because the `DecoderState` is parameterized over both `W` and `LZB` but the type `W` has to be used so the compiler doesn't complain.

If we declare the struct like so, we get an error:
```rust
pub struct DecoderState<LZB>
where
    LZB: lzbuffer::LZBuffer<W>
```
```
error[E0412]: cannot find type `W` in this scope
   --> src/decode/lzma.rs:103:29
    |
100 | pub struct DecoderState<LZB>
    |                            - help: you might be missing a type parameter: `, W`
...
103 |     LZB: lzbuffer::LZBuffer<W>,
    |                             ^ not found in this scope
```

If we add `W` we get an unused error:
```rust
pub struct DecoderState<W, LZB>
where
    W: io::Write,
    LZB: lzbuffer::LZBuffer<W>
```
```
error[E0392]: parameter `W` is never used
   --> src/decode/lzma.rs:100:25
    |
100 | pub struct DecoderState<W, LZB>
    |                         ^ unused parameter
    |
    = help: consider removing `W`, referring to it in a field, or using a marker such as `std::marker::PhantomData`
```